# 2020 US Data Exploration and Cleaning

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import plotly_express as px
import seaborn as sns

import warnings
warnings.simplefilter('ignore')
import statsmodels.formula.api as smf

## Purpose of this notebook

- initial cleaning and look at raw data files for happiness ratings in 2020 for US and COVID-19 cases in US
- create cleaned versions of files for analysis notebook

## Load datasets into dataframes

In [2]:
us_hap = pd.read_csv('../data/raw_data/2020/united_states.csv')
us_covid = pd.read_csv('../data/raw_data/2020/covid19_cases_us.csv')

## Start with US dataset for happiness

In [3]:
us_hap

RecordNo           endtime         state    qweek  i1_health  \
0             0  02/04/2020 08:11    California   week 1          5   
1             1  02/04/2020 08:12  Pennsylvania   week 1          3   
2             2  02/04/2020 08:15       Florida   week 1          1   
3             3  02/04/2020 08:15    California   week 1          5   
4             4  02/04/2020 08:15       Florida   week 1          1   
...         ...               ...           ...      ...        ...   
16977     16977  29/09/2020 04:57      Oklahoma  week 16          2   
16978     16978  29/09/2020 15:07    California  week 16          1   
16979     16979  29/09/2020 19:48  Rhode Island  week 16          4   
16980     16980  30/09/2020 04:17   Mississippi  week 16          1   
16981     16981  30/09/2020 15:26         Texas  week 16          5   

       i2_health  i7a_health                   i3_health  \
0              7           0              No, I have not   
1              4           0              No, I have not   
2             12           0              No, I have not   
3              8           1              No, I have not   
4              1           0              No, I have not   
...          ...         ...                         ...   
16977          1           1              No, I have not   
16978          1           0              No, I have not   
16979          0           0              No, I have not   
16980          0           1  Yes, and I tested negative   
16981          0           6                               

                           i4_health i5_health_1  ... w5_96 w5_99          w6  \
0                                                 ...                           
1                  No, they have not          No  ...                           
2                  No, they have not         Yes  ...                           
3                  No, they have not          No  ...                           
4                  No, they have not         Yes  ...                           
...                              ...         ...  ...   ...   ...         ...   
16977              No, they have not          No  ...    No    No         Yes   
16978              No, they have not          No  ...                      No   
16979              No, they have not          No  ...                      No   
16980  Yes, and they tested negative          No  ...              Don't Know   
16981                                             ...              Don't Know   

                      w7                        w9_1  \
0                                                      
1                                                      
2                                                      
3                                                      
4                                                      
...                  ...                         ...   
16977                  3            Some of the time   
16978         Don't know            Some of the time   
16979         Don't know                All the time   
16980                  4  More than half of the time   
16981  Prefer not to say                  At no time   

                             w9_2                        w9_3  \
0                                                               
1                                                               
2                                                               
3                                                               
4                                                               
...                           ...                         ...   
16977            Some of the time                  At no time   
16978            Some of the time                  At no time   
16979                All the time                All the time   
16980  More than half of the time  More than half of the time   
16981                  At no time                  At no time   

      

In [4]:
us_hap.shape


(16982, 323)

In [5]:
us_hap.columns

Index(['RecordNo', 'endtime', 'state', 'qweek', 'i1_health', 'i2_health',
       'i7a_health', 'i3_health', 'i4_health', 'i5_health_1',
       ...
       'w5_96', 'w5_99', 'w6', 'w7', 'w9_1', 'w9_2', 'w9_3', 'w9_4', 'w9_5',
       'disability'],
      dtype='object', length=323)

### Observations

16982 rows and 323 columns. Definitely not going to be using all of these columns and need to rename the columns based on the codebook.

## Selecting and renaming columns

In [6]:
us_hap_df=us_hap[['RecordNo', 'endtime','qweek','state','i3_health','d1_health_12','age','gender','WCRex2','CORE_B2_4','cantril_ladder','PHQ4_1','PHQ4_2','PHQ4_3','PHQ4_4','WCRex1','r1_1','r1_7','ox3_1','ox3_2','ox3_3','ox3_6','ox5_2','ox6','w2','w3','w4_1','w4_2','w4_7','w4_8','w4_9','w9_1','w9_2']]

In [7]:
col_dict = {
    'RecordNo':'record_no',
    'endtime':'date',
    'qweek': 'week',
    'i3_health':'covid_test',
    'd1_health_12':'mental_health_condition',
    'WCRex2':'confidence_NHS',
    'CORE_B2_4':'happiness_2wk_comparison',
    'cantril_ladder':'happiness_rating',
    'PHQ4_1':'little_interest',
    'PHQ4_2':'negative_feelings',
    'PHQ4_3':'anxiety',
    'PHQ4_4':'worry',
    'WCRex1':'gvt_rating',
    'r1_1':'perceptions_covid',
    'r1_7':'covid_effect',
    'ox3_1':'social_norms',
    'ox3_2':'social_expectations',
    'ox3_3':'appropriate_behaviors',
    'ox3_6':'follow_norms',
    'ox5_2':'trust_gvt',
    'ox6':'public_sector',
    'w2':'community_belonging',
    'w3':'belonging_change',
    'w4_1':'spoken_mental',
    'w4_2':'medication_mental',
    'w4_7':'seek_mental',
    'w4_8':'professional_mental',
    'w4_9':'readings_mental',
    'w9_1':'cheerful',
    'w9_2':'relaxed',}
us_hap_df = us_hap_df.rename(columns = col_dict)

In [8]:
us_hap_df.shape

(16982, 33)

In [9]:
us_hap_df.columns.unique

<bound method Index.unique of Index(['record_no', 'date', 'week', 'state', 'covid_test',
       'mental_health_condition', 'age', 'gender', 'confidence_NHS',
       'happiness_2wk_comparison', 'happiness_rating', 'little_interest',
       'negative_feelings', 'anxiety', 'worry', 'gvt_rating',
       'perceptions_covid', 'covid_effect', 'social_norms',
       'social_expectations', 'appropriate_behaviors', 'follow_norms',
       'trust_gvt', 'public_sector', 'community_belonging', 'belonging_change',
       'spoken_mental', 'medication_mental', 'seek_mental',
       'professional_mental', 'readings_mental', 'cheerful', 'relaxed'],
      dtype='object')>

### Observations

Initial filtering of the dataset to select certain columns. I have 16982 rows and 33 columns to work with

## Filtering columns further

In [10]:
cols_filter = ['date','week','state','mental_health_condition','age','gender','happiness_rating']

In [11]:
us_hap_filter=us_hap_df[cols_filter]

In [12]:
## happiness rating data unavailable for these weeks; drop rows
us_hap_filter = us_hap_filter.drop(us_hap_filter[us_hap_filter.week.isin(['week 1', 'week 2','week 3','week 4'])].index)

In [13]:
us_hap_filter['date']

4976     30/04/2020 16:33
4977     30/04/2020 16:31
4978     30/04/2020 16:32
4979     30/04/2020 16:34
4980     30/04/2020 16:34
               ...       
16977    29/09/2020 04:57
16978    29/09/2020 15:07
16979    29/09/2020 19:48
16980    30/09/2020 04:17
16981    30/09/2020 15:26
Name: date, Length: 12006, dtype: object

In [14]:
us_hap_filter['date'] = pd.to_datetime(us_hap_filter.date)
us_hap_filter['date'] = us_hap_filter['date'].dt.date


In [15]:
us_hap_filter

date     week         state mental_health_condition  age  gender  \
4976   2020-04-30   week 5       Georgia                      No   40    Male   
4977   2020-04-30   week 5    New Jersey                      No   62  Female   
4978   2020-04-30   week 5       Indiana                      No   61  Female   
4979   2020-04-30   week 5      Virginia                      No   60  Female   
4980   2020-04-30   week 5      Michigan                      No   37  Female   
...           ...      ...           ...                     ...  ...     ...   
16977  2020-09-29  week 16      Oklahoma                      No   27  Female   
16978  2020-09-29  week 16    California                      No   74  Female   
16979  2020-09-29  week 16  Rhode Island                      No   26  Female   
16980  2020-09-30  week 16   Mississippi                      No   22    Male   
16981  2020-09-30  week 16         Texas                      No   23    Male   

      happiness_rating  
4976                 2  
4977                 6  
4978                 9  
4979                 9  
4980                 3  
...                ...  
16977                4  
16978                5  
16979                5  
16980                8  
16981                0  

[12006 rows x 7 columns]

In [16]:
us_hap_filter.shape

(12006, 7)

### Observations

Final data to export contains 12006 rows and 7 columns that are relevant for analysis. Columns include date, week, state, happiness ratings, and some potentially related details about the person being surveyed.

## Exported to clean csv titled us_hap_cleaned.csv

In [17]:
# Making clean csv with these changes
us_hap_filter.to_csv('../data/cleaned_data/2020/us_hap_cleaned.csv')

## Now looking at COVID-19 cases data in US

In [18]:
us_covid.shape

(3340, 328)

In [19]:
us_covid.columns.unique

<bound method Index.unique of Index(['UID', 'iso2', 'iso3', 'code3', 'FIPS', 'Admin2', 'Province_State',
       'Country_Region', 'Lat', 'Long_',
       ...
       '11/24/20', '11/25/20', '11/26/20', '11/27/20', '11/28/20', '11/29/20',
       '11/30/20', '12/1/20', '12/2/20', '12/3/20'],
      dtype='object', length=328)>

### Observations

Need to select which columns are actually relevant to analysis based on codebook. Dates are provided in the columns (328 columns total) and 3340 rows. 

In [20]:
date_cols = us_covid.columns[11:]

In [21]:
### make dates into one column
us_covid= us_covid.melt(id_vars = us_covid.columns[6],
             value_vars = date_cols,
             var_name = 'date',
             value_name = 'cases')

In [22]:
us_covid['date']=pd.to_datetime(us_covid['date'])

In [23]:
us_covid

Province_State       date  cases
0              Alabama 2020-01-22      0
1              Alabama 2020-01-22      0
2              Alabama 2020-01-22      0
3              Alabama 2020-01-22      0
4              Alabama 2020-01-22      0
...                ...        ...    ...
1058775        Wyoming 2020-12-03   1693
1058776        Wyoming 2020-12-03   1149
1058777        Wyoming 2020-12-03      0
1058778        Wyoming 2020-12-03    497
1058779        Wyoming 2020-12-03    419

[1058780 rows x 3 columns]

### Observations

Final dataset contains the 3 columns of interest (location, date, and cases)

## Exported to clean csv titled us_covid.csv

In [24]:
# Making clean csv with these changes
us_covid.to_csv('../data/cleaned_data/2020/us_covid.csv')